src/s3/builders/list_objects.rs

```diff
@@ -0,0 +1,737 @@
+        self.extra_query_params = extra_query_params;
+        self
+    }
+
+    pub fn region(mut self, region: Option<String>) -> Self {
+        self.region = region;
+        self
+    }
+
+    pub fn delimiter(mut self, delimiter: Option<String>) -> Self {
+        self.delimiter = delimiter;
+        self
+    }
+
+    pub fn disable_url_encoding(mut self, disable_url_encoding: bool) -> Self {
+        self.disable_url_encoding = disable_url_encoding;
+        self
+    }
+
+    pub fn max_keys(mut self, max_keys: Option<u16>) -> Self {
+        self.max_keys = max_keys;
+        self
+    }
+
+    pub fn prefix(mut self, prefix: Option<String>) -> Self {
+        self.prefix = prefix;
+        self
+    }
+
+    pub fn key_marker(mut self, key_marker: Option<String>) -> Self {
+        self.key_marker = key_marker;
+        self
+    }
+
+    pub fn version_id_marker(mut self, version_id_marker: Option<String>) -> Self {
+        self.version_id_marker = version_id_marker;
+        self
+    }
+}
+
+/// Argument builder for
+/// [list_objects()](crate::s3::client::Client::list_objects) API.
+///
+/// Use the various builder methods to set parameters on the request. Finally to
+/// send the request and consume the results use the `ToStream` instance to get
+/// a stream of results. Pagination is automatically performed.
+#[derive(Clone, Debug, Default)]
+pub struct ListObjects {
+    client: Option<Client>,
+
+    // Parameters common to all ListObjects APIs.
+    extra_headers: Option<Multimap>,
+    extra_query_params: Option<Multimap>,
+    region: Option<String>,
+    bucket: String,
+    delimiter: Option<String>,
+    disable_url_encoding: bool,
+    max_keys: Option<u16>,
+    prefix: Option<String>,
+
+    // Options specific to ListObjectsV1.
+    marker: Option<String>,
+
+    // Options specific to ListObjectsV2.
+    start_after: Option<String>,
+    continuation_token: Option<String>,
+    fetch_owner: bool,
+    include_user_metadata: bool,
+
+    // Options specific to ListObjectVersions.
+    key_marker: Option<String>,
+    version_id_marker: Option<String>,
+
+    // Higher level options.
+    recursive: bool,
+    use_api_v1: bool,
+    include_versions: bool,
+}
+
+#[async_trait]
+impl ToStream for ListObjects {
+    type Item = ListObjectsResponse;
+
+    async fn to_stream(self) -> Box<dyn Stream<Item = Result<Self::Item, Error>> + Unpin + Send> {
+        if self.use_api_v1 {
+            let stream = ListObjectsV1::from(self).to_stream().await;
+            Box::new(stream.map(|v| v.map(|v| v.into())))
+        } else if self.include_versions {
+            let stream = ListObjectVersions::from(self).to_stream().await;
+            Box::new(stream.map(|v| v.map(|v| v.into())))
+        } else {
+            let stream = ListObjectsV2::from(self).to_stream().await;
+            Box::new(stream.map(|v| v.map(|v| v.into())))
+        }
```

Comment:
The `ToStream` instance for the lower level types is used in the `ToStream` instance for higher level `ListObjects` call.